### PR TITLE
Improve enums printing

### PR DESCRIPTION
**Pull request by nalimilan**
Print list of members and associated values when showing enum type,
as well as value and enum type when printing member.

Fixes https://github.com/JuliaLang/julia/issues/13138. Most of the credit is due to @KristofferC.

This gives:

``` julia
julia> @enum FRUIT apple=1 banana=10

julia> apple
apple = 1, a member of enum FRUIT

julia> FRUIT
Enum with 2 member values:
apple = 1
banana = 10
```

I must say that despite having read several issues, I still don't really understand the whole story with `show`, `showcompact` and `writemime`.

### CODE REVIEW

**Comment by StefanKarpinski** [comment]
This is too wordy and verbose for my taste.

**Comment by nalimilan** [comment]
@StefanKarpinski Any suggestions for an intermediate, shorter representation?

**Comment by nalimilan** [comment]
I can propose this, shorter and with less words:
```
julia> apple
apple::FRUIT = 1

julia> FRUIT
Enum FRUIT: apple = 1, banana = 10
```

Though for the first one, it would have been nice to mention "enum" somewhere.
For the second one, I'm not sure about the commas vs. line breaks to list the values.

**Comment by StefanKarpinski** [comment]
That does seem better.

**Comment by StefanKarpinski** [comment]
Tangentially, I'd like for `?x` to provide useful information about any object `x` – it's type, etc.

**Comment by JeffBezanson** [comment]
The story with display functions is that `writemime` should be used for verbose human-friendly representations, e.g. how the REPL shows things by default. `show` is used by `repr` and `string`. For example

```
julia> show(sin)
sin
julia> sin
sin (generic function with 13 methods)
```

Of course we'd all like to see a simpler design for printing at this point.

**Comment by nalimilan** [comment]
I've updated the branch to implement my last proposal, though with line breaks instead of simple commas: a real-world check with enums typically found in Base (especially in LibGit2) shows it's much more readable.

@JeffBezanson Thanks, that helps.

**Comment by KristofferC** [comment]
Cool, thanks for fixing it with the MIME stuff.
